opm/simulators/linalg/ParallelIstlInformation.hpp

```diff
@@ -44,6 +44,17 @@
 #include <dune/common/enumset.hh>
 #include <opm/common/utility/platform_dependent/reenable_warnings.h>
 
+#include <dune/common/version.hh>
+#include <dune/common/parallel/mpihelper.hh>
+
+namespace Opm::Parallel { 
+#if DUNE_VERSION_NEWER(DUNE_COMMON, 2, 7)
+    using Communication = Dune::Communication<Dune::MPIHelper::MPICommunicator>; 
+#else
+    using Communication = Dune::CollectiveCommunication<Dune::MPIHelper::MPICommunicator>;
+#endif
+} // end namespace Communication
+
```

Comment:
I have now seen this construct in multiple headers. Does that really work without compiler warnings?

Maybe the using statements in Opm::Parallel should get their own header and be defined only once. That header would need to get included elsewhere, of course. Seems more clean to me.